test/framework/deployment_helpers.go

```diff
@@ -245,3 +283,139 @@ func WaitForDNSUpgrade(ctx context.Context, input WaitForDNSUpgradeInput, interv
 		return false, nil
 	}, intervals...).Should(BeTrue())
 }
+
+type DeployUnevictablePodInput struct {
+	WorkloadClusterProxy               ClusterProxy
+	ControlPlane                       *controlplanev1.KubeadmControlPlane
+	IsDeploymentExisted                bool
+	WaitForDeploymentAvailableInterval []interface{}
+}
+
+func DeployUnevictablePod(ctx context.Context, input DeployUnevictablePodInput) {
+	workloadClient := input.WorkloadClusterProxy.GetClientSet()
+	numberOfDeployments := 3
+	if input.ControlPlane != nil {
+		untaintMasterNodesAndAddNodeSelectorToPod(ctx, untaintMasterNodesAndAddNodeSelectorToPodInput{
```

Comment:
Instead of untainting the nodes, what about adding a nodeselector selector targeting nodes with "node-role.kubernetes.io/master" and a toleration on the machine deployment?

the toleration should be similar to the one kubeadm adds to core DNS:

```
      tolerations:
      - key: CriticalAddonsOnly
        operator: Exists
      - key: node-role.kubernetes.io/master
        effect: NoSchedule
```

Reply:
Fixed. Thanks